src/Runner.Worker/Handlers/NodeScriptActionHandler.cs

```diff
@@ -137,13 +138,24 @@ public async Task RunAsync(ActionRunStage stage)
 
             if (Data.NodeVersion == "node12" && (ExecutionContext.Global.Variables.GetBoolean(Constants.Runner.Features.Node12Warning) ?? false))
             {
-                if (!ExecutionContext.JobContext.ContainsKey("Node12ActionsWarnings"))
+                var repoAction = Action as RepositoryPathReference;
+                var warningActions = new List<string>();
+                if (ExecutionContext.Global.Variables.TryGetValue("Node12ActionsWarnings", out var node12Warnings))
                 {
-                    ExecutionContext.JobContext["Node12ActionsWarnings"] = new ArrayContextData();
+                    warningActions = StringUtil.ConvertFromJson<List<string>>(node12Warnings);
```

Comment:
Maybe we could just append new warnings to the old ones instead of converting the old ones from/to json?